svm/src/transaction_processor.rs

```diff
@@ -120,6 +120,18 @@ pub struct TransactionProcessingConfig<'a> {
     pub limit_to_load_programs: bool,
     /// Recording capabilities for transaction execution.
     pub recording_config: ExecutionRecordingConfig,
+    /// Should failing transactions within the batch be dropped (no fee charged
+    /// & not committed).
+    pub drop_on_failure: bool,
+    /// If any transaction in the batch is not committed then the entire batch
+    /// should not be committed.
+    ///
+    /// # Note
+    ///
+    /// Without `drop_on_failure` this flag will still allow processed but
+    /// failing transactions to be committed. If both flags are set then any
+    /// failing transaction will cause all transactions to be aborted.
+    pub all_or_nothing: bool,
```

Comment:
maybe this should be one field with a three-value enum like `enum TransactionProcessingMode { Standard, DropOnFailure, AllOrNothing }` or similar? while testing i was surprised setting `all_or_nothing` true did nothing and thought `if config.all_or_nothing && processing_result.is_err() { ... }` was incorrect until i saw this comment